pennychat/views.py

```diff
@@ -1,3 +1,77 @@
-# from django.shortcuts import render
+from rest_framework import viewsets, mixins, generics
+from rest_framework.reverse import reverse
+from rest_framework.response import Response
+from rest_framework.status import HTTP_201_CREATED, HTTP_400_BAD_REQUEST
 
-# Create your views here.
+from users.models import UserProfile
+from .models import PennyChat, FollowUp
+from .serializers import PennyChatSerializer, FollowUpSerializer
+
+
+class PennyChatViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows users to be viewed or edited.
+    """
+    queryset = PennyChat.objects.all().order_by('-date')
+    serializer_class = PennyChatSerializer
+
+    def perform_create(self, serializer):
+        # get or create an anonymous user since we don't have auth
+        # this will eventually be replaced by user = self.request.user
+        user, created = UserProfile.objects.get_or_create(
+            real_name='Anonymous Profile',
+            defaults={
+                'email': 'anonymous@profile.com'
+            })
+        serializer.save(user=user)
+
+
+class ListCreateFollowUps(generics.GenericAPIView):
+    """
+    API endpoint that allows follow ups to be viewed or edited based on the foreign key of their associated chat.
+    """
+    serializer_class = FollowUpSerializer
+
+    def get(self, request, pk, format=None):
+        follow_ups = FollowUp.objects.filter(penny_chat_id=pk)
+        queryset = self.filter_queryset(follow_ups)
+
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        serializer = self.get_serializer(queryset, many=True)
```

Comment:
nit pick: `self.get_serializer(queryset, many=True)` this seems like it's doing something, but in reality it's just returning an empty `queryset`, right? Alternatives: Does this code work `self.get_serializer(None, many=True)`? If so you don't need the if clause. How about an empty list? `self.get_serializer([], many=True)` In that case you can replace line 39 with `page = self.paginate_queryset(queryset) or []` and remove the if clause. Finally if that doesn't work you can just do this:

```python
        page = self.paginate_queryset(queryset) or queryset
        serializer = self.get_serializer(page, many=True)
        return self.get_paginated_response(serializer.data)
```
just one less `if` clause to worry about

Reply:
You were right, no need for the if clause!